instrumentation/instasarama/consumer_group_handler_test.go

```diff
@@ -102,6 +123,18 @@ func TestConsumerGroupHandler_ConsumeClaim(t *testing.T) {
 			Access:  "consume",
 		}, span.Data.Kafka)
 
+		assert.Contains(t, h.Messages[1].Headers, &sarama.RecordHeader{
+			Key:   []byte("X_INSTANA_T"),
+			Value: []byte(span.TraceID),
+		})
+		assert.Contains(t, h.Messages[1].Headers, &sarama.RecordHeader{
+			Key:   []byte("X_INSTANA_S"),
+			Value: []byte(span.SpanID),
+		})
+		assert.Contains(t, h.Messages[1].Headers, &sarama.RecordHeader{
+			Key:   []byte("X_INSTANA_L_S"),
+			Value: []byte("1"),
+		})
```

Comment:
I'm entirely not sure about any of the changes in the tests for the *consumers*, in particular the cases where the original message has no headers and then we check in the test that headers have been added _at the consumer side_? That is different from what our other tracers do, which only add headers at the producer side, so this concept was new to me and a bit suprising. But from what I understand it is done so that the customers' applications can read the headers and continue the trace from the trace ID/span ID in the headers, is that correct?

If that understanding is correct, the header names and their contents are actually part of our public API in go and we need to make customers aware of this change, so they can update their application code somewhere between October 2022 and October 2023.